src/main/java/seedu/address/model/TaskList.java

```diff
@@ -32,7 +32,7 @@ public TaskList(ReadOnlyTaskList toBeCopied) {
 
     @Override
     public ObservableList<Task> getTaskList() {
-        return tasks;
+        return FXCollections.unmodifiableObservableList(tasks);
```

Comment:
Nice in identifying this mistake where `getTaskList()` should have returned an un-modifiable list.